### PR TITLE
feat: add extension `metal-agent`

### PR DESCRIPTION
**Pull request by utkuozdemir**
Add the new metal agent extension to run Talos in metal agent mode.

Related to https://github.com/siderolabs/talos/pull/9501.

Depends on https://github.com/siderolabs/talos-metal-agent/pull/1 and the initial release (`v0.1.0`) of the agent.

### CODE REVIEW

File: metal-agent-mode/pkg.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: metal-agent-mode
+variant: scratch
+shell: /toolchain/bin/bash
+dependencies:
+  - stage: base
```
Comment (utkuozdemir) [inline metal-agent-mode/pkg.yaml:5]: maybe accepts an image reference directly?

File: metal-agent-mode/vars.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+VERSION: "v1.0.0"
```
Comment (utkuozdemir) [inline metal-agent-mode/vars.yaml:1]: ref to pkg.yaml, configure renovate on it, use var everywhere

File: guest-agents/vars.yaml [hunk 3]
```diff
@@ -8,3 +8,5 @@ PCRE2_VERSION: 10.42
 XEN_GUEST_AGENT_VERSION: 0.4.0
 # renovate: datasource=github-releases depName=siderolabs/talos-vmtoolsd
 TALOS_VMTOOLSD_VERSION: v0.6.0
+# renovate: datasource=github-releases depName=siderolabs/talos-metal-agent
+TALOS_METAL_AGENT_VERSION: v0.1.0
```
Comment (utkuozdemir) [inline guest-agents/vars.yaml:12]: Will work only after we merge https://github.com/siderolabs/talos-metal-agent and release this version as the initial version.

File: guest-agents/metal-agent/manifest.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+version: v1alpha1
+metadata:
+  name: metal-agent
+  version: "$VERSION"
+  author: Sidero Labs
+  description: |
+    This system extension provides talos-metal-agent
+  compatibility:
+    talos:
+      version: ">= v1.8.0"
```
[the remark this answers is not in the document]
Comment (smira) [reply guest-agents/metal-agent/manifest.yaml:10]: it should be 1.9.0-alpha.0 at the very least, as it depends on Talos changes in v1.9

**Review by smira — APPROVED** [review]
let's also make a change to Image Factory to filter out this extension from the UI (only UI)

**Review by frezbo — APPROVED** [review]

**Pull request merged**